core/src/main/kotlin/xtdb/cache/MemoryCache.kt

```diff
@@ -9,17 +9,30 @@ import org.apache.arrow.memory.BufferAllocator
 import org.apache.arrow.memory.ForeignAllocation
 import org.apache.arrow.memory.util.MemoryUtil
 import xtdb.cache.PinningCache.IEntry
+import xtdb.util.isMetaFile
 import java.nio.ByteBuffer
 import java.nio.channels.ClosedByInterruptException
 import java.nio.channels.FileChannel
 import java.nio.file.Path
 import java.util.concurrent.CompletableFuture
 
-
 data class PathSlice(val path: Path, val offset: Long? = null, val length: Long? = null) {
     constructor(path: Path) : this(path, null, null)
 }
 
+data class MemoryCacheStats(
```

Comment:
rather than the double-up and the weakly-typed map here, I'd probably create something like

```kotlin
data class CacheSectionStats(sliceCount, weightBytes, pinned, unpinned)
```

then have one of those for meta and data?

```kotlin
data class MemoryCacheStats(pinningCacheStats, metaFileStats: CacheSectionStats, dataFileStats: CacheSectionStats)
```

(it's relatively unusual IME in Java/Kotlin to use Maps with keys that you know at compile-time)